File: executable/julia/grouped-bar.jl
using Plotly

using Plotly
Plotly.signin("test-runner", "9h29fe3l0x")

data = [
  [
    "x" => ["giraffes", "orangutans", "monkeys"], 
    "y" => [20, 14, 23], 
    "name" => "SF Zoo", 
    "type" => "bar"
  ], 
  [
    "x" => ["giraffes", "orangutans", "monkeys"], 
    "y" => [12, 18, 29], 
    "name" => "LA Zoo", 
    "type" => "bar"
  ]
]
layout = [
  "xaxis" => ["type" => "category"], 
  "categories" => ["giraffes", "orangutans", "monkeys"], 
  "barmode" => "group"
]

response = Plotly.plot([data], ["layout" => layout, "filename" => "grouped-bar", "fileopt" => "overwrite"])
plot_url = response["url"]

File: executable/r/grouped-bar.r
library(plotly)

p <- plotly(username='test-runner', key='9h29fe3l0x')

trace1 <- list(
  x = c("giraffes", "orangutans", "monkeys"), 
  y = c(20, 14, 23), 
  name = "SF Zoo", 
  type = "bar"
)
trace2 <- list(
  x = c("giraffes", "orangutans", "monkeys"), 
  y = c(12, 18, 29), 
  name = "LA Zoo", 
  type = "bar"
)

layout <- list(
  xaxis = list(type = "category"), 
  categories = c("giraffes", "orangutans", "monkeys"), 
  barmode = "group"
)



response <- p$plotly(trace0, trace1, kwargs=list(layout=layout, filename="grouped-bar", fileopt="overwrite"))
url <- response$url
filename <- response$filename